sdk/storage/azure-storage-blob/azure/storage/blob/aio/download_async.py

```diff
@@ -229,137 +130,241 @@ def __aiter__(self):
 
     async def __anext__(self):
         """Iterate through responses."""
-        if self._current_content is None:
-            if self.download_size == 0:
-                self._current_content = b""
-            else:
-                self._current_content = await process_content(
-                    self.response, self.initial_offset[0], self.initial_offset[1], self.encryption_options)
-            if not self._download_complete:
-                data_end = self.file_size
-                if self.length is not None:
-                    # Use the length unless it is over the end of the file
-                    data_end = min(self.file_size, self.length + 1)
-                self._iter_downloader = _AsyncChunkDownloader(
-                    client=self.clients.blob,
-                    non_empty_ranges=self.non_empty_ranges,
-                    total_size=self.download_size,
-                    chunk_size=self.config.max_chunk_get_size,
-                    current_progress=self.first_get_size,
-                    start_range=self.initial_range[1] + 1,  # start where the first download ended
-                    end_range=data_end,
-                    stream=None,
-                    parallel=False,
-                    validate_content=self.validate_content,
-                    encryption_options=self.encryption_options,
-                    use_location=self.location_mode,
-                    **self.request_options)
-                self._iter_chunks = self._iter_downloader.get_chunk_offsets()
-        elif self._download_complete:
+        if self._complete:
             raise StopAsyncIteration("Download complete")
+        if not self._iter_downloader:
+            # If no iterator was supplied, the download completed with
+            # the initial GET, so we just return that data
+            self._complete = True
+            return self._current_content
+
+        if not self._iter_chunks:
+            self._iter_chunks = self._iter_downloader.get_chunk_offsets()
```

Comment:
Good question :)
Yes I think we do need the else here, because this iterator object is always created _after_ we have downloaded the first chunk of data - i.e. we already have a certain amount of content to yield before we start pulling down additional chunks.

The `self._iter_chunks` will only be `None` on the first iteration, which means we have not yet yielded that initial data. So on the first iteration we yield the first chunk will initializing subsequent chunks. Make sense?

Reply:
Ah that makes things clear; thanks++, I was assuming the iter_chunks flow was distinct from the prior removed bit that set up _current_content.